### PR TITLE
Fix artifact uploads to GCS

### PR DESCRIPTION
**Pull request by tobim**
### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [x] check that gsutil allows specifying a full path to the destination

### CODE REVIEW

File: .github/workflows/vast.yaml [hunk 1]
```diff
@@ -379,7 +379,7 @@ jobs:
       - name: Upload Artifact to GCS
         if: ${{ github.event_name == 'push' || github.event_name == 'release' }}
         run: |
-          gsutil -m cp "$BUILD_DIR/$PACKAGE_NAME.tar.gz" "gs://${{ secrets.GCS_BUCKET }}"
+          gsutil -m cp "${{ env.BUILD_DIR }}/package/${{ env.PACKAGE_NAME }}.tar.gz" "gs://${{ secrets.GCS_BUCKET }}/$PACKAGE_NAME.tar.gz"
```
Comment (tobim) [inline .github/workflows/vast.yaml:382]: I'm not sure whether we have any downstream jobs that depend on that name, so maybe we can even keep the filename from cpack?

Comment (dominiklohmann) [reply .github/workflows/vast.yaml:382]: The proprietary plugin CI depends on it, but you can easily adjust that to use another name.

Comment (dominiklohmann) [inline .github/workflows/vast.yaml:382]:
```suggestion
          gsutil -m cp "${{ env.BUILD_DIR }}/package/${{ env.PACKAGE_NAME }}.tar.gz" "gs://${{ secrets.GCS_BUCKET }}/${{ env.PACKAGE_NAME }}.tar.gz"
```

Let's just make this consistent.

**Review by dominiklohmann — APPROVED** [review]

File: .github/workflows/vast.yaml (hunk 1 above)
Comment (tobim) [reply .github/workflows/vast.yaml]: Let's do that some other time.

**Pull request merged**